inheritance/__init__.py

```diff
@@ -0,0 +1,51 @@
+import check50
+import check50.c
+import re
+
+@check50.check()
+def exists():
+    """inheritance.c exists"""
+    check50.exists("inheritance.c")
+
+@check50.check(exists)
+def compiles():
+    """inheritance.c compiles"""
+    check50.c.compile("inheritance.c", lcs50=True)
+
+@check50.check(exists)
+def compiles():
+    """inheritance compiles"""
+    check50.c.compile("inheritance.c", lcs50=True)
+    inheritance = re.sub("int\s+main", "int distro_main", open("inheritance.c").read())
+    testing = open("testing.c").read()
+    with open("inheritance_test.c", "w") as f:
+        f.write(inheritance)
+        f.write("\n")
+        f.write(testing)
+    check50.c.compile("inheritance_test.c", lcs50=True)
+
+@check50.check(compiles)
+def frees_memory():
```

Comment:
Let's do this check last, since I imagine most students will try to get the right output first and then worry about handling any memory leaks.